docs/blog/_posts/2018-03-05-seventh-dotty-milestone-release.md

```diff
@@ -0,0 +1,157 @@
+---
+layout: blog-page
+title: Announcing Dotty 0.6.0 and 0.7.0-RC1
+author: Allan Renucci
+authorImg: /images/allan.jpg
+date: 2018-03-05
+---
+
+Today, we are excited to release Dotty versions 0.6.0 and 0.7.0-RC1. These releases
+serve as a technology preview that demonstrates new language features and the compiler supporting them.
+
+If you’re not familiar with Dotty, it's a platform to try out new language concepts and compiler
+technologies for Scala. The focus is mainly on simplification. We remove extraneous syntax
+(e.g. no XML literals), and try to boil down Scala’s types into a smaller set of more fundamental
+constructs. The theory behind these constructs is researched in
+[DOT](https://infoscience.epfl.ch/record/215280), a calculus for dependent object types.
+You can learn more about Dotty on our [website](http://dotty.epfl.ch).
+
+<!--more-->
+
+This is our seventh scheduled release according to our [6-week release schedule](/docs/usage/version-numbers.html).
+The [previous technology preview](https://github.com/lampepfl/dotty/releases/tag/0.6.0-RC1) focussed
+on bug fixes and stability work.
+
+## What’s new in the 0.7.0-RC1 technology preview?
+
+### Enums Simplicification [#4003](https://github.com/lampepfl/dotty/pull/4003)
+The previously introduced syntax and rules for enum were arguably too complex. We can considerably
+simplify them by taking away one capability: that cases can have bodies which can define members.
+Arguably, if we choose an ADT decomposition of a problem, it's good style to write all methods using
+pattern matching instead of overriding individual cases. So this removes an unnecessary choice.
+We now treat enums unequivocally as classes. They can have methods and other statements just like
+other classes can. Cases in enums are seen as a form of constructors. We do not need a
+distinction between enum class and enum object anymore. Enums can have companion objects just like
+normal classes can, of course.
+
+Let's consider how `Option` can be represented as an enum. Previously using an enum class:
+```scala
+enum class Option[+T] {
+   def isDefined: Boolean
+}
+object Option {
+  case Some[+T](x: T) {
+     def isDefined = true
+  }
+  case None {
+     def isDefined = false
+  }
+
+  def apply[T(x: T): Option[T] = if (x == null) None else Some(x)
+}
+```
+
+And now:
+```scala
+enum Option[+T] {
+  case Some(x: T)
+  case None
+
+  def isDefined: Boolean = this match {
+    case None => false
+    case some => true
+  }
+}
+
+object Option {
+  def apply[T](x: T): Option[T] = if (x == null) None else Some(x)
+}
+```
+
+You can visit our website for more information about [enumerations](/docs/reference/enums/enums.html)
+and how we can use them to model [Algebraic Data Types](/docs/reference/enums/adts.html).
+
+### Unused Parameters [#3342](https://github.com/lampepfl/dotty/pull/3342)
```

Comment:
@nicolasstucki Do you want to do this?

Reply:
@nicolasstucki  Probably we should say a couple words about going from phantom types to unused parameters (https://github.com/lampepfl/dotty/issues/2040#issuecomment-369628306), probably in the release notes rather than in the docs for the latest release (to avoid confusing new users). Starting point:

> TL;DR: With unused parameters you can mark values of arbitrary types to be erased at runtime, instead of having to implement separate phantom types.